fix: do not scan local modules

thisislawatts opened the pull request:
Do not treat local modules as if they are 3rd party packages hosted on npm.

Reference: #11 

This PR addresses the above need in two ways:

* Validates that a folder/file does not exist locally before assuming it is an external package.
* Introduces a new configuration option to ignore paths based on a Regular Expression

thisislawatts committed:
fix: do not scan local modules

Do not treat local modules as if they are 3rd party
packages hosted on npm.

thisislawatts committed:
fix: supports eg/file.js

The NodeJS require behaviour will also load a file if it's
exists in the file directory

thisislawatts committed:
feat: adds setting to ignore based on paths

Introduces a configuration option to ignore a path based on a regex.

erwinw commented on src/getImports/babelParser.js:
```diff
@@ -26,13 +30,39 @@ export function getPackages(fileName, source, language) {
+      const target = path.dirname(fileName) + path.sep + node.source.value;
+      const fileExists =
+        fs.existsSync(target) ||
+        fs.existsSync(target + '.js') ||
```
Please make the extensions configurable - think of `.jsx`, `.ts`, etc. Thanks!

thisislawatts replied:
Excellent point @erwinpleo, we already expose `typescriptExtensions` and `javascriptExtensions` as configuration options so we should reuse these

thisislawatts replied:
Proposed implementation here:
https://github.com/snyk/vulncost/pull/16/commits/927e53ccbc5a4213ac3057d718a9bcda80b245cb

erwinw commented on src/getImports/babelParser.js:
```diff
@@ -26,13 +30,39 @@ export function getPackages(fileName, source, language) {
+        pathIgnored = new RegExp(path).test(node.source.value);
+
+        if (pathIgnored) {
+          continue;
```
Isn’t the intention to break out of the `for` loop here?
```suggestion
          break;
```

thisislawatts committed:
feat: use js|typescript config regex

Reuse JS/Typescript configuration options
to validate the found files rather than hard coding
file extensions.

bmvermeer reviewed (approved):
Works great!

bmvermeer commented:
need to add entry to changelog and merge

thisislawatts committed:
chore: format

bmvermeer committed:
Update CHANGELOG.md

update CHANGELOG

thisislawatts commented:
:tada: This PR is included in version 1.2.0 :tada:

The release is available on [GitHub release](https://github.com/snyk/vulncost/releases/tag/v1.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
